internal/server/singleprocess/service_instance.go

```diff
@@ -18,8 +20,40 @@ func (s *service) ListInstances(
 	var err error
 	switch scope := req.Scope.(type) {
 	case *pb.ListInstancesRequest_DeploymentId:
-		result, err = s.state.InstancesByDeployment(scope.DeploymentId, nil)
+		// For more complex deployments, sometimes it takes a bit longer for
+		// the instance to respond. This blocks on a response when querying
+		// a deployment by id, and lets the user know it is taking a while
+		//
+		// The default case is no timeout, no blocking. Make a request and return
 
+		if req.WaitTimeout != "" {
+			connectTimeout, err := time.ParseDuration(req.WaitTimeout)
+			if err != nil {
+				return nil, err
```

Comment:
This `err` is gonna be a Go error, we should probably wrap this in a status:

```go
status.Errorf(codes.FailedPrecondition, "Error parsing wait_timeout: %s", err)
```

We're not like super consistent about this but its a good idea when we recognize it.